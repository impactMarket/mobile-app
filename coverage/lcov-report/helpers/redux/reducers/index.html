
<!doctype html>
<html lang="en">

<head>
    <title>Code coverage report for helpers/redux/reducers</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <link rel="shortcut icon" type="image/x-icon" href="../../../favicon.png" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
    
<body>
<div class='wrapper'>
    <div class='pad1'>
        <h1><a href="../../../index.html">All files</a> helpers/redux/reducers</h1>
        <div class='clearfix'>
            
            <div class='fl pad1y space-right2'>
                <span class="strong">35.48% </span>
                <span class="quiet">Statements</span>
                <span class='fraction'>22/62</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">24.39% </span>
                <span class="quiet">Branches</span>
                <span class='fraction'>10/41</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">100% </span>
                <span class="quiet">Functions</span>
                <span class='fraction'>5/5</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">35.48% </span>
                <span class="quiet">Lines</span>
                <span class='fraction'>22/62</span>
            </div>
        
            
        </div>
        <p class="quiet">
            Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
        </p>
    </div>
    <div class='status-line low'></div>
    <div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file low" data-value="app.ts"><a href="app.ts.html">app.ts</a></td>
	<td data-value="36.36" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 36%"></div><div class="cover-empty" style="width: 64%"></div></div>
	</td>
	<td data-value="36.36" class="pct low">36.36%</td>
	<td data-value="11" class="abs low">4/11</td>
	<td data-value="22.22" class="pct low">22.22%</td>
	<td data-value="9" class="abs low">2/9</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="36.36" class="pct low">36.36%</td>
	<td data-value="11" class="abs low">4/11</td>
	</tr>

<tr>
	<td class="file medium" data-value="auth.ts"><a href="auth.ts.html">auth.ts</a></td>
	<td data-value="66.67" class="pic medium">
	<div class="chart"><div class="cover-fill" style="width: 66%"></div><div class="cover-empty" style="width: 34%"></div></div>
	</td>
	<td data-value="66.67" class="pct medium">66.67%</td>
	<td data-value="6" class="abs medium">4/6</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="4" class="abs medium">2/4</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="66.67" class="pct medium">66.67%</td>
	<td data-value="6" class="abs medium">4/6</td>
	</tr>

<tr>
	<td class="file empty" data-value="index.ts"><a href="index.ts.html">index.ts</a></td>
	<td data-value="0" class="pic empty">
	<div class="chart"><div class="cover-fill" style="width: 0%"></div><div class="cover-empty" style="width: 100%"></div></div>
	</td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	</tr>

<tr>
	<td class="file low" data-value="modalDonate.ts"><a href="modalDonate.ts.html">modalDonate.ts</a></td>
	<td data-value="36.36" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 36%"></div><div class="cover-empty" style="width: 64%"></div></div>
	</td>
	<td data-value="36.36" class="pct low">36.36%</td>
	<td data-value="11" class="abs low">4/11</td>
	<td data-value="22.22" class="pct low">22.22%</td>
	<td data-value="9" class="abs low">2/9</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="36.36" class="pct low">36.36%</td>
	<td data-value="11" class="abs low">4/11</td>
	</tr>

<tr>
	<td class="file medium" data-value="stories.ts"><a href="stories.ts.html">stories.ts</a></td>
	<td data-value="57.14" class="pic medium">
	<div class="chart"><div class="cover-fill" style="width: 57%"></div><div class="cover-empty" style="width: 43%"></div></div>
	</td>
	<td data-value="57.14" class="pct medium">57.14%</td>
	<td data-value="7" class="abs medium">4/7</td>
	<td data-value="40" class="pct low">40%</td>
	<td data-value="5" class="abs low">2/5</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="57.14" class="pct medium">57.14%</td>
	<td data-value="7" class="abs medium">4/7</td>
	</tr>

<tr>
	<td class="file low" data-value="user.ts"><a href="user.ts.html">user.ts</a></td>
	<td data-value="22.22" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 22%"></div><div class="cover-empty" style="width: 78%"></div></div>
	</td>
	<td data-value="22.22" class="pct low">22.22%</td>
	<td data-value="27" class="abs low">6/27</td>
	<td data-value="14.29" class="pct low">14.29%</td>
	<td data-value="14" class="abs low">2/14</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="22.22" class="pct low">22.22%</td>
	<td data-value="27" class="abs low">6/27</td>
	</tr>

</tbody>
</table>
</div>
                <div class='push'></div><!-- for sticky footer -->
            </div><!-- /wrapper -->
            <div class='footer quiet pad2 space-top1 center small'>
                Code coverage generated by
                <a href="https://istanbul.js.org/" target="_blank">istanbul</a>
                at Sun Jun 20 2021 16:32:25 GMT+0100 (Western European Summer Time)
            </div>
        </div>
        <script src="../../../prettify.js"></script>
        <script>
            window.onload = function () {
                prettyPrint();
            };
        </script>
        <script src="../../../sorter.js"></script>
        <script src="../../../block-navigation.js"></script>
    </body>
</html>
    