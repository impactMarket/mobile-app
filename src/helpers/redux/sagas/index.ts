import { all, spawn } from 'typed-redux-saga';

import { startWatchingNetworkConnectivity } from './offline';
import stories from './stories';

export default function* rootSagas() {
    return yield all([stories]);
    // return yield all([spawn(startWatchingNetworkConnectivity), stories]);
}
