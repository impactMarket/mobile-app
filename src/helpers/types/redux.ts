import { ContractKit } from '@celo/contractkit';
import { Subscription } from '@unimodules/core';
import {
    SET_USER_CELO_INFO,
    SET_USER_INFO,
    SET_USER_EXCHANGE_RATE,
    // INIT_USER,
    SET_USER_WALLET_BALANCE,
    SET_USER_IS_BENEFICIARY,
    SET_USER_IS_BLOCKED,
    SET_USER_IS_SUSPECT,
    SET_USER_IS_COMMUNITY_MANAGER,
    SET_CELO_KIT,
    SET_APP_BENEFICIARY_HAS_ACCEPTED_TERMS,
    SET_APP_MANAGER_HAS_ACCEPTED_TERMS,
    SET_COMMUNITY_CONTRACT,
    SET_COMMUNITY,
    RESET_USER_APP,
    // RESET_NETWORK_APP,
    SET_PUSH_NOTIFICATION_TOKEN,
    SET_AUTH_TOKEN,
    SET_USER_LANGUAGE,
    SET_APP_SUSPECT_WRONG_DATETIME,
    SET_APP_FROM_WELCOME_SCREEN,
    SET_EXCHANGE_RATES,
    appAction,
    modalDonateAction,
    storiesAction,
} from 'helpers/constants';

import { ICommunitiesListStories, ICommunityStory } from './endpoints';
import { CommunityAttributes, UserAttributes } from './models';
import { IUserWallet } from './state';

// action
interface UserWalletAction {
    type: typeof SET_USER_CELO_INFO;
    payload: IUserWallet;
}

interface UserMetadataAction {
    type: typeof SET_USER_INFO;
    payload: UserAttributes;
}

interface UserExchangeRateAction {
    type: typeof SET_USER_EXCHANGE_RATE;
    payload: number;
}

interface UserSetBalanceAction {
    type: typeof SET_USER_WALLET_BALANCE;
    payload: string;
}

interface UserSetIsBeneficiaryAction {
    type: typeof SET_USER_IS_BENEFICIARY;
    payload: boolean;
}

interface UserSetIsBlockedAction {
    type: typeof SET_USER_IS_BLOCKED;
    payload: boolean;
}

interface UserSetIsSuspectAction {
    type: typeof SET_USER_IS_SUSPECT;
    payload: boolean;
}

interface UserSetIsCommunityManagerAction {
    type: typeof SET_USER_IS_COMMUNITY_MANAGER;
    payload: boolean;
}

interface UserSetBeneficiaryAcceptedRulesAction {
    type: typeof SET_APP_BENEFICIARY_HAS_ACCEPTED_TERMS;
    payload: boolean;
}

interface UserSetManagerAcceptedRulesAction {
    type: typeof SET_APP_MANAGER_HAS_ACCEPTED_TERMS;
    payload: boolean;
}

interface CeloKitAction {
    type: typeof SET_CELO_KIT;
    payload: ContractKit;
}

interface SetCommunityContractAction {
    type: typeof SET_COMMUNITY_CONTRACT;
    payload: any;
}

interface SetCommunityMetadataAction {
    type: typeof SET_COMMUNITY;
    payload: CommunityAttributes;
}

interface ResetUserAction {
    type: typeof RESET_USER_APP;
    payload: any;
}

interface SetTokenPushNotificationsAction {
    type: typeof SET_PUSH_NOTIFICATION_TOKEN;
    payload: string;
}

interface SetAuthTokenAction {
    type: typeof SET_AUTH_TOKEN;
    payload: string;
}
interface UserLanguageAction {
    type: typeof SET_USER_LANGUAGE;
    payload: string;
}

interface SetAppSuspectWrongDateTime {
    type: typeof SET_APP_SUSPECT_WRONG_DATETIME;
    payload: {
        suspect: boolean;
        timeDiff: number;
    };
}

interface SetAppFromWelcomeScreen {
    type: typeof SET_APP_FROM_WELCOME_SCREEN;
    payload: string;
}

interface SetAppEchangeRatesAction {
    type: typeof SET_EXCHANGE_RATES;
    payload: any;
}

interface OpenModalDonateAction {
    type: typeof modalDonateAction.OPEN;
    payload: CommunityAttributes;
}

interface GoToConfirmModalDonateAction {
    type: typeof modalDonateAction.GO_TO_CONFIRM_DONATE;
    payload: {
        inputAmount: string;
        amountInDollars: number;
        backNBeneficiaries: number;
        backForDays: number;
    };
}

interface GoToErrorModalDonateAction {
    type: typeof modalDonateAction.GO_TO_ERROR_DONATE;
}

interface GoBackToModalDonateAction {
    type: typeof modalDonateAction.GO_BACK_TO_DONATE;
}

interface InProgressModalDonateAction {
    type: typeof modalDonateAction.IN_PROGRESS;
    payload: boolean;
}

interface CloseModalDonateAction {
    type: typeof modalDonateAction.CLOSE;
}

interface InitLoadStoriesActionRequest {
    type: typeof storiesAction.INIT_REQUEST;
    payload: { start: number; end: number };
}

interface InitLoadStoriesActionSuccess {
    type: typeof storiesAction.INIT_SUCCESS;
    payload: ICommunitiesListStories[];
}

interface InitLoadStoriesActionFailure {
    type: typeof storiesAction.INIT_FAILURE;
}

interface LoadMyStoriesActionRequest {
    type: typeof storiesAction.USER_STORIES_REQUEST;
    payload: ICommunityStory[];
}

interface LoadMyStoriesActionSuccess {
    type: typeof storiesAction.USER_STORIES_SUCCESS;
    payload: ICommunityStory[];
}

interface LoadMyStoriesActionFailure {
    type: typeof storiesAction.USER_STORIES_FAILURE;
}

interface LoadMoreStoriesAction {
    type: typeof storiesAction.CONCAT;
    payload: ICommunitiesListStories[];
}

interface SetAppPushNotificationListeners {
    type: typeof appAction.SET_PUSH_NOTIFICATION_LISTENERS;
    payload: {
        notificationReceivedListener: Subscription;
        notificationResponseReceivedListener: Subscription;
    };
}

export type UserActionTypes =
    | UserWalletAction
    | UserSetBalanceAction
    | UserSetIsBeneficiaryAction
    | UserSetIsBlockedAction
    | UserSetIsSuspectAction
    | UserSetIsCommunityManagerAction
    | ResetUserAction
    | UserMetadataAction
    | UserLanguageAction
    | UserExchangeRateAction
    | SetCommunityContractAction
    | SetCommunityMetadataAction;

export type AuthActionTypes =
    | SetTokenPushNotificationsAction
    | SetAuthTokenAction;

export type AppActionTypes =
    | CeloKitAction
    | SetAppEchangeRatesAction
    | SetAppSuspectWrongDateTime
    | SetAppFromWelcomeScreen
    | SetAppPushNotificationListeners
    | UserSetBeneficiaryAcceptedRulesAction
    | UserSetManagerAcceptedRulesAction;

export type ModalActionTypes =
    | OpenModalDonateAction
    | GoToConfirmModalDonateAction
    | GoToErrorModalDonateAction
    | GoBackToModalDonateAction
    | InProgressModalDonateAction
    | CloseModalDonateAction;

export type StoriesActionTypes =
    | InitLoadStoriesActionRequest
    | InitLoadStoriesActionSuccess
    | InitLoadStoriesActionFailure
    | LoadMoreStoriesAction
    | LoadMyStoriesActionRequest
    | LoadMyStoriesActionSuccess
    | LoadMyStoriesActionFailure;

export type IStoreCombinedActionsTypes =
    | UserActionTypes
    | AuthActionTypes
    | AppActionTypes
    | ModalActionTypes
    | StoriesActionTypes;
