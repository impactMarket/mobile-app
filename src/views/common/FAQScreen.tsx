import React, { useState } from 'react';
import {
    StyleSheet,
    ScrollView,
    View,
    Image,
} from 'react-native';
import * as Linking from 'expo-linking';
import { useNavigation } from '@react-navigation/native';
import Header from '../../components/Header';
import { List, Text } from 'react-native-paper';
import { iptcColors } from '../../helpers';


function FAQScreen() {
    const navigation = useNavigation();
    const [expanded, setExpanded] = useState<string[]>([]);

    const handlePress = (tag: string) => {
        let newExpanded: string[];
        if (expanded.indexOf(tag) === -1) {
            newExpanded = [...expanded];
            newExpanded.push(tag);
        } else {
            newExpanded = expanded.filter((e) => e !== tag);
        }
        setExpanded(newExpanded);
    }
    return (
        <>
            <Header
                title="FAQ"
                navigation={navigation}
                hasBack={true}
            />
            <ScrollView style={styles.contentView}>

                <List.Accordion
                    title="Where can I claim?"
                    expanded={expanded.indexOf('where_can_i_claim') !== -1}
                    onPress={() => handlePress('where_can_i_claim')}
                >
                    <View style={styles.descriptionView}>
                        <Text style={styles.descriptionText}>Lorem ipsum dolor sit amet, consectetur adipiscing elit. Etiam finibus neque id urna ultricies, vitae auctor purus vulputate. Maecenas viverra augue ac ex iaculis convallis. Praesent nec orci convallis leo tristique facilisis nec nec mi. <Text style={{ color: iptcColors.softBlue }} onPress={() => Linking.openURL('https://google.pt')}>Etiam vestibulum risus maximus</Text> ornare pretium. Nunc posuere ipsum ac aliquam luctus. Aenean scelerisque tellus vel imperdiet pulvinar. Sed facilisis nisi sit amet augue mattis imperdiet. Nulla molestie ut arcu at dapibus. Nunc dictum nisi est, nec vulputate purus placerat dictum.</Text>
                        <Image
                            style={{ width: 50, height: 50 }}
                            source={{
                                uri: 'https://reactnative.dev/img/tiny_logo.png',
                            }}
                        />
                        <Text style={styles.descriptionText}>Mauris ac eros in enim scelerisque commodo. Suspendisse potenti. Vivamus vel dolor enim. Interdum et malesuada fames ac ante ipsum primis in faucibus. Curabitur hendrerit vehicula vestibulum.</Text>
                    </View>
                </List.Accordion>
                <List.Accordion
                    title="Do I need to buy crypto to join a community?"
                    expanded={expanded.indexOf('need_to_buy_crypto_to_join_community') !== -1}
                    onPress={() => handlePress('need_to_buy_crypto_to_join_community')}
                >
                    <View style={styles.descriptionView}>
                        <Text style={styles.descriptionText}>Lorem ipsum dolor sit amet, consectetur adipiscing elit. Etiam finibus neque id urna ultricies, vitae auctor purus vulputate. Maecenas viverra augue ac ex iaculis convallis. Praesent nec orci convallis leo tristique facilisis nec nec mi.</Text>
                    </View>
                </List.Accordion>
                <List.Accordion
                    title="How can I communicate with my community?"
                >
                </List.Accordion>
                <List.Accordion
                    title="How do I use the money I claimed?"
                >
                </List.Accordion>
                <List.Accordion
                    title="Are communities wallets insured?"
                >
                </List.Accordion>
                <List.Accordion
                    title="How can I leave my community?"
                >
                </List.Accordion>

            </ScrollView>
        </>
    );
}

const styles = StyleSheet.create({
    contentView: {
        paddingHorizontal: 20,
    },
    descriptionView: {
        paddingHorizontal: 20
    },
    descriptionText: {
        opacity: 0.84,
        fontFamily: "Gelion-Regular",
        fontSize: 18,
        fontWeight: "normal",
        fontStyle: "normal",
        lineHeight: 25,
        letterSpacing: 0,
        color: "#1e3252"
    }
});

export default FAQScreen;