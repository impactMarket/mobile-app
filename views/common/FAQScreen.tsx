import React, { useState } from 'react';
import {
    StyleSheet,
    ScrollView,
} from 'react-native';
import { useNavigation } from '@react-navigation/native';
import Header from '../../components/Header';
import { List, Text } from 'react-native-paper';


function FAQScreen() {
    const navigation = useNavigation();

    const [expanded, setExpanded] = useState(true);

    const handlePress = () => setExpanded(!expanded);
    return (
        <>
            <Header
                title="FAQ"
                navigation={navigation}
                hasBack={true}
            />
            <ScrollView style={styles.contentView}>

                <List.Accordion
                    title="Where can I claim?"
                    expanded={expanded}
                    onPress={handlePress}>
                    <Text>Lorem ipsum dolor sit amet, consectetur adipiscing elit. Etiam finibus neque id urna ultricies, vitae auctor purus vulputate. Maecenas viverra augue ac ex iaculis convallis. Praesent nec orci convallis leo tristique facilisis nec nec mi. Etiam vestibulum risus maximus ornare pretium. Nunc posuere ipsum ac aliquam luctus. Aenean scelerisque tellus vel imperdiet pulvinar. Sed facilisis nisi sit amet augue mattis imperdiet. Nulla molestie ut arcu at dapibus. Nunc dictum nisi est, nec vulputate purus placerat dictum.</Text>
                </List.Accordion>
                <List.Accordion
                    title="Do I need to buy crypto to join a community?"
                >
                </List.Accordion>
                <List.Accordion
                    title="How can I communicate with my community?"
                >
                </List.Accordion>
                <List.Accordion
                    title="How do I use the money I claimed?"
                >
                </List.Accordion>
                <List.Accordion
                    title="Are communities wallets insured?"
                >
                </List.Accordion>
                <List.Accordion
                    title="How can I leave my community?"
                >
                </List.Accordion>

            </ScrollView>
        </>
    );
}

const styles = StyleSheet.create({
    contentView: {
        marginHorizontal: 20,
    },
});

export default FAQScreen;