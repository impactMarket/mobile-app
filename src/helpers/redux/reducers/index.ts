import { combineReducers } from 'redux';

import { appReducer } from './app';
import { authReducer } from './auth';
import { userReducer } from './user';
import { modalDonateReducer } from './view';

export default combineReducers({
    user: userReducer,
    auth: authReducer,
    app: appReducer,
    modalDonate: modalDonateReducer,
});
