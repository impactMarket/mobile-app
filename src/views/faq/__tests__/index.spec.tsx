import combinedReducer from 'helpers/redux/reducers';
import React from 'react';
import { Provider } from 'react-redux';
import { create } from 'react-test-renderer';
import { createStore } from 'redux';

import FAQScreen from '../index';

const store = createStore(combinedReducer);

const tree = create(
    <Provider store={store}>
        <FAQScreen />
    </Provider>
).toJSON();

describe('FAQ screen test', () => {
    test('should render FAQScreen correctly', () => {
        expect(tree).toMatchSnapshot();
    });
});
