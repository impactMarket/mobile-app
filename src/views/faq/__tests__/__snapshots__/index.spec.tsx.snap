// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FAQ screen test suite should render FAQ screen correctly 1`] = `ShallowWrapper {}`;
